great_expectations/data_context/store/datasource_store.py

```diff
@@ -137,6 +161,23 @@ def set_by_name(
         )
         self.set(datasource_key, datasource_config)
 
+    def create(
+        self, datasource_config: DatasourceConfig
+    ) -> Optional[GeCloudResourceRef]:
```

Comment:
Do these somewhat remove the need for `crud_by_name` methods? Should we rename this to be consistent (perhaps `create_by_config`)? I'm okay with how it is currently but I'd like to create a uniform approach for subsequent types.

Reply:
I think so - I was thinking about removing the `crud_by_name` in favor of passing in a config which will now have `name` and `id_` and building the appropriate key based on the backend. I was hoping to get the other CRUD methods working in a subsequent PR within this same ticket after getting Create working. Does that work?

Reply:
Not sure why I can't reply to your other comment, but this is in reference to your comment about `build_key` being abstract or on the parent. I thought about this, but decided against it since it is not needed for other the other backends and the signature will be different. I think there is room for improvement here if we think of a better idea but I think this can work for now and at least moves the responsibility to the right class.